src/modules/python/pbs_v1_module_init.c

```diff
@@ -0,0 +1,815 @@
+		return PyErr_Format(PyExc_Exception,
+			 "get_fullhostname() failed to acqiure the server host name");
+	}
+
+	if ((job_attr_idx = cr_attrdef_idx(job_attr_def, JOB_ATR_LAST)) == NULL) {
+		return PyErr_Format(PyExc_Exception,
+			"Failed creating job attribute search index");
+	}
+	if ((node_attr_idx = cr_attrdef_idx(node_attr_def, ND_ATR_LAST)) == NULL) {
+		return PyErr_Format(PyExc_Exception,
+			"Failed creating node attribute search index");
+	}
+	if ((que_attr_idx = cr_attrdef_idx(que_attr_def, QA_ATR_LAST)) == NULL) {
+		return PyErr_Format(PyExc_Exception,
+			"Failed creating queue attribute search index");
+	}
+	if ((svr_attr_idx = cr_attrdef_idx(svr_attr_def, SVR_ATR_LAST)) == NULL) {
+		return PyErr_Format(PyExc_Exception,
+			"Failed creating server attribute search index");
+	}
+	if ((sched_attr_idx = cr_attrdef_idx(sched_attr_def, SCHED_ATR_LAST)) == NULL) {
+		return PyErr_Format(PyExc_Exception,
+			"Failed creating sched attribute search index");
+	}
+	if ((resv_attr_idx = cr_attrdef_idx(resv_attr_def, RESV_ATR_LAST)) == NULL) {
+		return PyErr_Format(PyExc_Exception,
+			"Failed creating resv attribute search index");
+	}
+	if (cr_rescdef_idx(svr_resc_def, svr_resc_size) != 0) {
+		return PyErr_Format(PyExc_Exception,
+			"Failed creating resc definition search index");
+	}
+
+	/* initialize the pointers in the resource_def array */
+	for (i = 0; i < (svr_resc_size - 1); ++i) {
+		svr_resc_def[i].rs_next = &svr_resc_def[i+1];
+	}
+
+	/* set python interp data */
+	svr_interp_data.init_interpreter_data = NULL;
+	svr_interp_data.destroy_interpreter_data = NULL;
+	svr_interp_data.interp_started = 1;
+	svr_interp_data.pbs_python_types_loaded = 0;
+	if (gethostname(svr_interp_data.local_host_name, PBS_MAXHOSTNAME) == -1) {
+		return PyErr_Format(PyExc_Exception,
+			"gethostname() failed to acquire the local host name");
+	}
+	svr_interp_data.daemon_name = strdup(MODULE_NAME);
+	svr_interp_data.data_initialized = 1;
+
+	/* construct _pbs_v1 module */
+	module = pbs_v1_module_init();
+	if (module == NULL) {
+		return PyErr_Format(PyExc_Exception,
+			 PBS_PYTHON_V1_MODULE_EXTENSION_NAME
+			 " module initialization failed");
+	}
+
+	/*
+	 * get a borrowed reference to sys.modules and add our module to it in order
+	 * to prevent an import cycle while loading the PBS python types
+	 */
+	py_sys_modules = PyImport_GetModuleDict();
+	if (PyDict_SetItemString(py_sys_modules,
+			PBS_PYTHON_V1_MODULE_EXTENSION_NAME, module)) {
+		return PyErr_Format(PyExc_Exception,
+			"failed to addd the " PBS_PYTHON_V1_MODULE_EXTENSION_NAME
+			" module to sys.modules");
+	}
+
+	/* load python types into the _pbs_v1 module */
+	if ((pbs_python_load_python_types(&svr_interp_data) == -1)) {
+		PyDict_DelItemString(py_sys_modules,
+			PBS_PYTHON_V1_MODULE_EXTENSION_NAME);
+		return PyErr_Format(PyExc_Exception,
+			"pbs_python_load_python_types() failed to load Python types");
+	}
+
+	return module;
+}
+
+/*
```

Comment:
The following "dummy" variable declarations and function definitions are the same as the ones in the standalone src/tools/pbs_python.c. Please merge them somehow, so that there'll be only one place to update or add to, when new functions are introduced.

Reply:
I agree.  And my apologies for the delay in responding.  I was on vacation for a couple of weeks and then playing catch-up.  I will strip out the common code as much as seems reasonable and push an update once that is done.

Reply:
I attempted to address the issue you raised about code duplication with the following commit: f54836830e25d5d8efa3a9845f74336c747b0a37.  Please let me know if you these changes are acceptable.